src/sdam/description/server.rs

```diff
@@ -82,6 +83,30 @@ impl Default for ServerType {
     }
 }
 
+/// Struct modeling the `topologyVersion` field included in the server's hello and legacy hello
+/// responses.
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct TopologyVersion {
+    pub(crate) process_id: ObjectId,
+    pub(crate) counter: i64,
+}
+
+impl TopologyVersion {
+    pub(crate) fn is_more_recent_than(&self, existing_tv: TopologyVersion) -> bool {
+        self.process_id != existing_tv.process_id || self.counter > existing_tv.counter
+    }
+}
+
+impl From<TopologyVersion> for Bson {
```

Comment:
This is used instead of `serde` so we can convert to BSON without having to handle errors (this is called in `hello_command`, which currently is infallible).